### ISSUE TITLE
Too much memory use with all architectures

### ISSUE THREAD

**Issue by alexvaca0** [idealo/image-super-resolution#184]
Hi, even changing the example notebook like this: 

```{python}
lr_train_patch_size = 150
layers_to_extract = [5, 9]
scale = 4
hr_train_patch_size = lr_train_patch_size * scale

rrdn  = RRDN(arch_params={'C':4, 'D':3, 'G':32, 'G0':32, 'T': 1, 'x':scale}, patch_size=lr_train_patch_size)
f_ext = Cut_VGG19(patch_size=hr_train_patch_size, layers_to_extract=layers_to_extract)
discr = Discriminator(patch_size=hr_train_patch_size, kernel_size=3)
```

it uses always 15300MB of GPU Memory, even when I lower T to 1. But if I use the default parameters you have, with T = 10, which should use more memory, uses exactly the same.

Can you please confirm which tensorflow, cuda, cudnn versions you have? Has anyone else found this issue?

**Comment by Ristellise** [comment]
Hey there! I'm also having this issue as well when trying to train on GPU, so I more or elss gave up and decided to use CPU. `RTX 3060` running a `Ryzen 5 3600` with `48 GB` of memory.

**Comment by mortium91** [comment]
Yeah i have the same problem! I run it as a docker container, but it consumes all the gpu memory from all the rest of the containers.

I know it should be possible to limit it in TF:
https://stackoverflow.com/questions/34199233/how-to-prevent-tensorflow-from-allocating-the-totality-of-a-gpu-memory/34200194#34200194

However I import ISR as a lib, so havent figured out yet if this is possible in that case..
